py/server/deephaven/table.py

```diff
@@ -1343,6 +1345,76 @@ def __init__(self, j_partitioned_table):
         self._constituent_changes_permitted = None
         self._is_refreshing = None
 
+    @classmethod
+    def from_partitioned_table(cls,
+                               table: Table,
+                               key_cols: Union[str, List[str]] = None,
+                               unique_keys: bool = None,
+                               constituent_column: str = None,
+                               constituent_table_columns: List[Column] = None,
+                               constituent_changes_permitted: bool = None) -> PartitionedTable:
+        """Creates a PartitionedTable from the provided underlying partitioned Table.
+
+        Note: key_cols, unique_keys, constituent_table_columns, constituent_table_columns,
+        constituent_changes_permitted must either be all None or all have values
+
+        Args:
+            table (Table): the underlying partitioned table
+            key_cols (Union[str, List[str]]): the key column name(s) of 'table'
+            unique_keys (bool): whether the keys in 'table' are guaranteed to be unique
+            constituent_column (str): the constituent column name in 'table'
+            constituent_table_columns (list[Column]): the column definitions of the constituent table
```

Comment:
I think both methods need to port more of the guidance from the Java version about requiring non-empty input in `table` if we need to infer the other args.